lib/ioc-volumes

```diff
@@ -0,0 +1,196 @@
+#!/bin/sh
+
+__print_exit () {
+    echo "$1"
+    __stop_jail $fulluuid no_unmount
+    exit 1
+}
+
+# format of $volumes: /path/to/some/host/dir[:/location/in/jail/root[:mount,options]]
+# multiple volumes can be space separated
+__mount_volumes () {
+    local _uuid=$1
+    local _mountpoint _source _dest _full_dest _mount_options
+    local _mounts _mounted _retval
+    local volumes="$(__get_jail_prop volumes ${_uuid})"
+
+    if [ -z "${volumes}" -o "${volumes}" == "none" ] ; then
+        return
+    fi
+
+    _mountpoint="$(__get_jail_prop mountpoint ${_uuid})"
+
+    for _volume in ${volumes} ; do
+        _source=""
+        _dest=""
+        _mount_opions=""
+
+        if [ -z "$(echo "${_volume}" | grep ':')" ] ; then
+            _source="${_volume}"
+        else
+            # split into source, dest and mount options
+            _source=$(echo "${_volume}" | cut -d: -f1)
+            _dest=$(echo "${_volume}" | cut -d: -f2)
+            _mount_options=$(echo "${_volume}" | cut -d: -f3)
+        fi
+
+        # set destination path to same as source if missing
+        if [ -z "${_dest}" ] ; then
+            _dest="${_source}"
+        fi
+
+        # add mointpoint
+        _full_dest="${_mountpoint}/root${_dest}"
+
+        # set default mount options if missing
+        if [ -z "${_mount_options}" ] ; then
+            _mount_options="rw"
+        fi
+
+        # source path checking
+        if [ -z "${_source}" ] ; then
+            __print_exit "  ERROR: volume source cannot be empty: ${_source}"
```

Comment:
no, and if you read the code you would understand why. this also stops the jail before exiting. i renamed it `__stop_and_die` to be more obvious though.